test/realm_logger_test.dart

```diff
@@ -0,0 +1,337 @@
+      results.addEntries([MapEntry(rootIsolateName, rootLogsCount)]);
+      return results;
+    });
+
+    expect(results[isolate1Name], 1);
+    expect(results[rootIsolateName], 1);
+  });
+}
+
+Future<List<LogRecord>> _attachToLoggerBeforeAction(String isolateName, FutureOr<void> Function() action, {Level? logLevel, bool setNewLogger = false}) async {
+  final completer = Completer<void>();
+  List<LogRecord> events = [];
+  await Future<void>.delayed(Duration(milliseconds: 200));
+  if (setNewLogger) {
+    Realm.logger = Logger.detached(generateRandomString(10))
+      ..level = logLevel ?? RealmLogLevel.info
+      ..onRecord.listen((event) {
+        events.add(event);
+      });
+  } else {
+    if (logLevel != null) {
+      Realm.logger.level = logLevel;
+    }
+    Realm.logger.onRecord.listen((event) {
+      events.add(event);
+    });
+  }
+
+  try {
+    await action();
+    await completer.future.timeout(Duration(seconds: 2), onTimeout: () => throw Exception("Stop waiting for logs"));
+  } catch (error) {
+    completer.complete();
+  }
+
+  return events;
+}
+
+Future<List<LogRecord>> _attachToLoggerAfterAction(String isolateName, FutureOr<void> Function() action, {Level? logLevel, bool setNewLogger = false}) async {
+  final completer = Completer<void>();
+  await Future<void>.delayed(Duration(milliseconds: 200));
+  List<LogRecord> events = [];
+  try {
+    await action();
+    if (setNewLogger) {
+      Realm.logger = Logger.detached(generateRandomString(10))
+        ..level = logLevel ?? RealmLogLevel.info
+        ..onRecord.listen((event) {
+          events.add(event);
+        });
+    } else {
+      if (logLevel != null) {
+        Realm.logger.level = logLevel;
+      }
+      Realm.logger.onRecord.listen((event) {
+        events.add(event);
+      });
+    }
+
+    await completer.future.timeout(Duration(seconds: 2), onTimeout: () => throw Exception("Stop waiting for logs"));
+  } catch (error) {
+    completer.complete();
+  }
+  return events;
+}
+
+Future<void> _simulateLogs(String message) async {
+  RealmInternal.log(RealmLogLevel.trace, message);
+  RealmInternal.log(RealmLogLevel.debug, message);
+  RealmInternal.log(RealmLogLevel.detail, message);
+  RealmInternal.log(RealmLogLevel.info, message);
+  RealmInternal.log(RealmLogLevel.fatal, message);
+  RealmInternal.log(RealmLogLevel.error, message);
+  RealmInternal.log(RealmLogLevel.warn, message);
+}
+
+Map<Level, int> _matchedMessagesCountPerLevel(List<LogRecord>? logRecords, String specificMessage) {
+  Map<Level, int> matchedMessages = {};
+  for (Level level in Level.LEVELS) {
+    final matches = logRecords?.where((element) => element.level == level && element.message.contains(specificMessage));
+    matchedMessages.addEntries([MapEntry(level, matches?.length ?? 0)]);
+  }
+  return matchedMessages;
+}
+
+void _expectLogMessages(
+  Map<Level, int> actualMessages, {
+  List<Level> notExpectedMessagesFromLevels = const [],
```

Comment:
This API is not great and hard to read. `expectLogMessages` has an argument of `notExpectedMessages`.
Why we need to expect counts when there is not even an argument for count in this function.
This is really making me stop and think how we are validating and what we are validating.

Reply:
It is really not clear to me how I will be able to write a new test using this common functions

Reply:
It would be nice to have one comon function for validation for all the levels